### ISSUE TITLE
Texconv W/H switches - Decrease/Increase only?

### ISSUE THREAD

**Issue by JT8D-17** [Microsoft/DirectXTex#80]
More a question about the documentation, I figure, but I coudln't find any other place where to post this.

As I understand it, adding -w [value] and -h [value] to the texconv command will resize the input image to the values specified in those switches. Issuing only -w [value] OR -h [value] requires the -pow2 switch to maintain the original image's aspect ratio.
In both cases, an input image will be resized to the output width/height regardless of input width/height (greater or smaller than the output value).
Is this correct?

If it is, what about adding -wmin/-hmin and -wmax/-hmax switches to future Texconv releases to limit resizing operations to increases (-wmin/-hmin) or decreases (-wmax/-hmax) in image dimensions only?

(Sideline question: Is Lanczos filtering supported for resizing operations? It's at least not mentioned in the -if switch documentation.)

At the moment, I have to run input images through NConvert to resize them to a maximum width/height (if they're too large) and then use Texconv to compress them into BC1 or 2 or 3 format (Nconvert only supports BC3). I'd like to eliminate NConvert from the process to streamline it a bit.

**Comment by walbourn** [comment]
The ``-w`` and ``-h`` switches are intended to set specific sizes, so they are less useful in large batch operations unless you invoke the tool once per texture. It doesn't really have any sense of aspect ratio and just takes you are your word. If you only specify one, it keeps the original value of the other.

The ``-pow2`` switch is intended to automate size selection for batch operations by forcing the texture size to a power of 2 based on whatever size the original image is at.  ``-pow2`` does try to maintain aspect ratio. The maximum size for a ``-pow2`` operation is based on the [Direct3D Hardware Feature Level](https://blogs.msdn.microsoft.com/chuckw/2012/06/20/direct3d-feature-levels/) you specify with ``-fl``:

    { L"9.1",  2048 },
    { L"9.2",  2048 },
    { L"9.3",  4096 },
    { L"10.0", 8192 },
    { L"10.1", 8192 },
    { L"11.0", 16384 },
    { L"11.1", 16384 },
    { L"12.0", 16384 },
    { L"12.1", 16384 },

It defaults to 16384 (the 'largest minimum' for Direct3D Feature Level 11.0 or better).

So in other words this would resize everything in the current directory to a power of 2 that fit within the 4096 size limit of Direct3D Feature Level 9.3 (Shader Model 2.0 / 3.0 parts)

> texconv -pow2 -fl 9.3 *.bmp

**Comment by JT8D-17** [comment]
In my usage case, I am indeed invoking Texconv on a per-texture basis from a batch script because it's the only way to get the texture back where it came from.
The "-fl" switch, however, is not of much use in my case since it's limited to a minimum of  2048 pixel and I want to work cases where output size has to be lower than that - 1024 or even 512.

But thanks for the clarification and quick reply. Much appreciated!
